core/src/autogluon/core/models/ensemble/bagged_ensemble_model.py

```diff
@@ -262,13 +273,16 @@ def predict_proba(self, X, normalize=None, **kwargs):
     def _predict_proba(self, X, normalize=False, **kwargs):
         return self.predict_proba(X=X, normalize=normalize, **kwargs)
 
-    def score_with_oof(self, y):
+    def score_with_oof(self, y, weights=None):
         self._load_oof()
         valid_indices = self._oof_pred_model_repeats > 0
         y = y[valid_indices]
         y_pred_proba = self.oof_pred_proba[valid_indices]
-
-        return self.score_with_y_pred_proba(y=y, y_pred_proba=y_pred_proba)
+        if weights is None:
```

Comment:
Can be simplified to avoid code duplication:
```
if weights is not None:
    weights = weights[valid_indices]
return self.score_with_y_pred_proba(y=y, y_pred_proba=y_pred_proba, weights=weights)
```